curve25519/src/lib.rs

```diff
@@ -23,21 +23,11 @@ pub struct Error;
 
 /// This trait is implemented by the backing implementations.
 /// Only used for implementation agility.
-trait Curve25519 {
+pub trait Curve25519 {
```

Comment:
Do we actually need this public? If I recall correctly, this was only used internally to ensure different implementations have the same interface.

Reply:
I made this private again and added a comment for why we have this here.